observatory-dags/observatory/dags/telescopes/jstor.py

```diff
@@ -0,0 +1,396 @@
+                      'Chrome/89.0.4389.90 Safari/537.36',
+        'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,'
+                  'application/signed-exchange;v=b3;q=0.9'
+
+    }
+    return headers
+
+
+def create_gmail_service() -> Resource:
+    """ Build the gmail service.
+
+    :return: Gmail service instance
+    """
+    gmail_api_conn = BaseHook.get_connection(AirflowConns.GMAIL_API)
+    scopes = ['https://www.googleapis.com/auth/gmail.readonly', 'https://www.googleapis.com/auth/gmail.modify']
+    creds = Credentials.from_authorized_user_info(gmail_api_conn.extra_dejson, scopes=scopes)
+
+    service = build('gmail', 'v1', credentials=creds, cache_discovery=False)
+
+    return service
+
+
+def get_label_id(service: Resource, label_name: str) -> str:
+    """ Get the id of a label based on the label name.
+
+    :param service: Gmail service
+    :param label_name: The name of the label
+    :return: The label id
+    """
+    existing_labels = service.users().labels().list(userId='me').execute()['labels']
+    label_id = [label['id'] for label in existing_labels if label['name'] == label_name]
+    if label_id:
+        label_id = label_id[0]
+    else:
+        # create label
+        label_body = {
+            'name': label_name,
+            'messageListVisibility': 'show',
+            'labelListVisibility': 'labelShow',
+            'type': 'user'
+        }
+        result = service.users().labels().create(userId='me', body=label_body).execute()
+        label_id = result['id']
+    return label_id
+
+
+def message_has_label(message: dict, label_id: str) -> bool:
+    """ Checks if a message has the given label
+
+    :param message: A Gmail message
+    :param label_id: The label id
+    :return: True if the message has the label
+    """
+    label_ids = message['labelIds']
+    for label in label_ids:
+        if label == label_id:
+            return True
+
+
+def list_available_releases(service: Resource, publisher: str, processed_label_id: str) -> Dict[Pendulum, List[dict]]:
+    """ List the available releases by going through the messages of a gmail account and looking for a specific pattern.
+
+    If a message has been processed previously it has a specific label, messages with this label will be skipped.
+    The message should include a download url. The head of this download url contains the filename, from which the
+    release date and publisher can be derived.
+
+    :param service: Gmail service
+    :param publisher: Name of the publisher
+    :param processed_label_id: Id of the 'processed_reports' label
+    :return: Dictionary with release dates as key and reports info as value, where reports info is a list of country
+    and/or institution reports.
+    """
+
+    available_releases = {}
+    # Call the Gmail API
+    results = service.users().messages().list(userId='me', q='subject:"JSTOR Publisher Report Available"').execute()
```

Comment:
```suggestion
    results = service.users().messages().list(userId='me', q='subject:"JSTOR Publisher Report Available"', labelIds=["INBOX"]).execute()
```

This is so that the sent items are not included.